src/main/java/frc/trigon/robot/poseestimation/robotposesources/RobotPoseSource.java

```diff
@@ -76,11 +67,57 @@ public double getLastResultTimestamp() {
         return inputs.lastResultTimestamp;
     }
 
-    private Pose2d getUnCachedRobotPose() {
-        final Pose3d cameraPose = doubleArrayToPose3d(inputs.cameraPose);
+    public Matrix<N3, N1> calculateStdDevs() {
+        if (inputs.distanceFromBestTag < RobotPoseSourceConstants.MAXIMUM_DISTANCE_FROM_TAG_FOR_PNP_METERS)
+            return solvePNPAverageDistanceToStdDevs();
+        return assumedHeadingAverageDistanceToStdDevs();
+    }
+
+    /**
+     * If the robot is close enough to the tag, it gets the estimated solve PNP pose.
+     * If it's too far, it assumes the robot's heading and calculates its position from there.
+     *
```

Comment:
Write about how assumedHeading is more robust, but we also need solvePNP to reset the heading